torch/_inductor/decomposition.py

```diff
@@ -680,3 +680,25 @@ def put(self, index, source, accumulate=False):
 def put_(self, index, source, accumulate=False):
     out = aten.put(self, index, source, accumulate=accumulate)
     return self.copy_(out)
+
+
+@register_decomposition(aten.index_reduce)
+def index_reduce(self, dim: int, index, src, reduction_type: str, **kwargs):
+    return index_reduce_(self.clone(), dim, index, src, reduction_type, **kwargs)
```

Comment:
Is this decomposition decomposing a functional op with a inplace op? Would it be desirable to decompose it with a functional op as well?